control-plane/gateways/metadata.go

```diff
@@ -0,0 +1,117 @@
+package gateways
+
+import (
+	"golang.org/x/exp/slices"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
+)
+
+const labelManagedBy = "mesh.consul.hashicorp.com/managed-by"
+
+var defaultLabels = map[string]string{labelManagedBy: "consul-k8s"}
+
+// Annotations returns the computed set of annotations for a resource by inheriting
+// from the MeshGateway's annotations and adding specified key-values from the
+// GatewayClassConfig based on the Kubernetes object type.
+func (b *meshGatewayBuilder) Annotations(object client.Object) map[string]string {
```

Comment:
what are your thoughts on having separate `AnnotationsForX` methods (and the others) rather than the switch to make calling a bit more intuitive? like it reads a little strange to me to pass in an empty object at the call site to figure out what labels/annotations are needed

Reply:
That sounds perfectly cool to me!